### PR TITLE
Global Priority Filter for Violations Overview/Outline

### PR DESCRIPTION
**Pull request by pkrall520**
This turns the filter on the violations overview into a global filter. When the user changes that filter it will then filter the markers on the file, tree view, and violations outline. It will also automatically run a check code on the open files.

New parts of the code:
- There is a new PriorityUtil which allows use to access the the currently active priorities
- Added a static runCodeReviewOnFiles on the ReviewCodeCmd, so I didn't have to copy and paste the command to run the review code
- Added getOpenFiles to the plugin to pulls all the files the user has open in the editor, so I can refresh them.

Sorry about the couple of files that got auto formatted. If it is an issue, I can revert them and reapply the changes.

### CODE REVIEW

File: net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/ui/preferences/RuleTableViewerSorter.java [hunk 1]
```diff
@@ -148,9 +148,9 @@ public void setSortDescending(boolean sortDescending) {
     /**
      * @return Returns the comparator.
      */
-    public Comparator<Rule> getComparator() {
-        return comparator;
-    }
+//    public Comparator<Rule> getComparator() {
```
Comment (adangel) [inline net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/ui/preferences/RuleTableViewerSorter.java]: Please keep this in. I know, in eclipse this is highlighted as an error, but I think, the problem is due to incorrect eclipse API configuration. I'm planning to increase the minimum required eclipse version, this plugins runs with, and hopefully this issue will be fixed then.

File: net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/ui/views/actions/PriorityFilterAction.java [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
 import net.sourceforge.pmd.eclipse.ui.priority.PriorityDescriptor;
 import net.sourceforge.pmd.eclipse.ui.views.PriorityFilter;
 import net.sourceforge.pmd.eclipse.ui.views.ViolationOutline;
 import net.sourceforge.pmd.eclipse.ui.views.ViolationOverview;
-
-import org.eclipse.jface.action.Action;
-import org.eclipse.jface.resource.ImageDescriptor;
-import org.eclipse.jface.viewers.ViewerFilter;
-import org.eclipse.swt.widgets.Display;
+import scala.tools.jline_embedded.internal.Log;
```
Comment (adangel) [inline net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/ui/views/actions/PriorityFilterAction.java]: Please use `org.apache.log4j.Logger`.
You'll need to add a private static class member then:

```
private static final Logger LOG = Logger.getLogger(PriorityFilterAction.class);
```

File: net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/ui/views/actions/PriorityFilterAction.java [hunk 3]
```diff
@@ -151,6 +160,13 @@ public void run() {
             outlineView.refresh();
         else if (overviewView != null)
             overviewView.refresh();
+        
+        /* Get all the opened files and tell them to run a "review code" on the file */
+        try {
+			ReviewCodeCmd.runCodeReviewOnFiles(PMDPlugin.getDefault().getOpenFiles());
+		} catch (CommandException e) {
+			Log.error(e);
```
Comment (adangel) [inline net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/ui/views/actions/PriorityFilterAction.java]: And then use here `LOG.error(e);`.

File: net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/util/PriorityUtil.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ * update.
+ * 
+ * @author Phillip Krall
+ *
+ */
+public class PriorityUtil {
+
+	private static PriorityFilter priorityFilter = new PriorityFilter();
+	static {
+		UISettings.currentPriorities(true);
```
Comment (adangel) [inline net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/util/PriorityUtil.java]: I think, this call should not be needed.

Comment (pkrall520) [reply net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/util/PriorityUtil.java]: Yeah you are right about this. I was grabbing the values and caching before. I must have missed this with the clean up.

**Comment by pkrall520** [comment]
@adangel I thought about the priority filter being on a package level and I think from a coding perspective we can easily do it, but from a user perspective, I don't think it is a good idea. I don't think it will be intuitive enough to keep changing the filter on the violations overview from package to package. Plus I think that shows all the packages, so it probably its a good idea for this..

Maybe a future thought would be to add a dropdown to the eclipse toolbar for something like this. or just add it to the package preferences? But then it wouldn't be in their face, so that doesn't seem good either

**Comment by adangel** [comment]
@pkrall520 I think long term, it needs to go somehow to the project settings. If has multiple projects opened and changes the filter on e.g. the violation outline - what's the expectation here? If first thought, that the change should not affect all projects - but I've just seen, that in general these views (e.g. the Problems view) show issues from all projects (unless you add a filter to show only problems of the current in-focus project). So, this might indeed be a workspace setting?

Storing such settings doesn't happen automatically IMHO - you'll always need to persist the settings explicitely (see e.g. `PreferencesManagerImpl`).

I think, we'll just need to decide:
* Should changing this priority filter affect all projects?
* Should the filter settings be persisted across restarts of eclipse. And if yes, is it a workspace preference (global) or a per-project preference?

Anyway, we don't need to decide now. I'll merge this PR and we can go from there.

**Comment by adangel** [comment]
@pkrall520 About the code style: I've finished now the checkstyle cleanup. We are using spaces only for indentation and no tabs. You can import the formatter options for eclipse from https://github.com/pmd/build-tools/tree/master/config/eclipse . The checkstyle configuration, that is used by pmd-eclipse-plugin is here: https://github.com/pmd/pmd-eclipse-plugin/blob/master/pmd-checkstyle-config.xml - it's almost the same as for PMD itself.

**Comment by adangel** [comment]
@pkrall520 Thanks for the PR - it's merged! :tada:
A new build is available at: https://dl.bintray.com/pmd/pmd-eclipse-plugin/snapshots/updates/

**Pull request merged**